test/functional/legacy/listchars_spec.lua

```diff
@@ -0,0 +1,96 @@
+-- Tests for 'listchars' display with 'list' and :list.
+
+local helpers = require('test.functional.helpers')
+local feed, insert, source = helpers.feed, helpers.insert, helpers.source
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+
+describe("'listchars'", function()
+  before_each(clear)
+
+  it("is working with 'list'", function()
```

Comment:
Mentioning for the future: I suggest leaving out "is working" in the test descriptions; `"with 'list'"` is sufficient. Trying to fit in the clever BDD "it" format just forces us to repeat this "is working" filler.